File: app/src/main/java/app/rental/utsavstha/rentalapp/Utils/PreferencesManager.java
package app.rental.utsavstha.rentalapp.Utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import app.rental.utsavstha.rentalapp.RentApplication;

/**
 * Created by utsavstha on 8/23/17.
 */

public class PreferencesManager {
    private static PreferencesManager singleton = null;
    private SharedPreferences preferences = null;

    private PreferencesManager(Context c){
        preferences = PreferenceManager.getDefaultSharedPreferences(c);
    }

    public static PreferencesManager getPreferences(){
        if( singleton == null){
            singleton = new PreferencesManager(RentApplication.getContext());
        }
        return singleton;
    }

    public void setPreference(String key, Object value) {
        // The SharedPreferences editor - must use commit() to submit changes
        SharedPreferences.Editor editor = preferences.edit();
        if(value instanceof Integer )
            editor.putInt(key, ((Integer) value).intValue());
        else if (value instanceof String)
            editor.putString(key, (String)value);
        else if (value instanceof Boolean)
            editor.putBoolean(key, (Boolean)value);
        else if (value instanceof Long)
            editor.putLong(key, (Long)value);
        editor.commit();
    }

    public int getInt(String key, int defaultValue) {
        return preferences.getInt(key, defaultValue);
    }

    public String getString(String key, String defaultValue) {
        return preferences.getString(key, defaultValue);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return preferences.getBoolean(key, defaultValue);
    }

    public long getLong(String key, long defaultValue) {
        return preferences.getLong(key, defaultValue);
    }
}

File: app/src/main/java/app/rental/utsavstha/rentalapp/Utils/PreferencesManager.kt
package app.rental.utsavstha.rentalapp.Utils

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

import app.rental.utsavstha.rentalapp.RentApplication

/**
 * Created by utsavstha on 8/23/17.
 */

class PreferencesManager private constructor(c: Context) {
    private var preferences: SharedPreferences? = null

    init {
        preferences = PreferenceManager.getDefaultSharedPreferences(c)
    }

    fun setPreference(key: String, value: Any) {
        // The SharedPreferences editor - must use commit() to submit changes
        val editor = preferences!!.edit()
        if (value is Int)
            editor.putInt(key, value.toInt())
        else if (value is String)
            editor.putString(key, value)
        else if (value is Boolean)
            editor.putBoolean(key, value)
        else if (value is Long)
            editor.putLong(key, value)
        editor.commit()
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return preferences!!.getInt(key, defaultValue)
    }

    fun getString(key: String, defaultValue: String): String? {
        return preferences!!.getString(key, defaultValue)
    }

    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return preferences!!.getBoolean(key, defaultValue)
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return preferences!!.getLong(key, defaultValue)
    }

    companion object {
        private var singleton: PreferencesManager? = null

        fun getPreferences(): PreferencesManager {
            if (singleton == null) {
                singleton = PreferencesManager(RentApplication.context!!)
            }
            return singleton as PreferencesManager
        }
    }
}
